sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -152,15 +152,12 @@ public virtual Task<Response<BulkRegistryOperationResponse>> CreateIdentitiesWit
                     Authentication = x.Key.Authentication,
                     Capabilities = x.Key.Capabilities,
                     DeviceScope = x.Key.DeviceScope,
-                    ParentScopes = x.Key.ParentScopes,
                     Status = string.Equals(ExportImportDeviceStatus.Disabled.ToString(), x.Key.Status?.ToString(), StringComparison.OrdinalIgnoreCase)
                                 ? ExportImportDeviceStatus.Disabled
                                 : ExportImportDeviceStatus.Enabled,
                     StatusReason = x.Key.StatusReason,
-                    Tags = x.Value.Tags,
-                    Properties = new PropertyContainer(x.Value.Properties?.Desired, x.Value.Properties?.Reported),
                     ImportMode = ExportImportDeviceImportMode.Create
-                });
+                }.WithTags(x.Value.Tags).WithPropertiesFrom(x.Value.Properties).WithParentScopes(x.Key.ParentScopes));
```

Comment:
So this is because these fields, being collections, are readonly by default, and so they need additional helper methods? 
https://github.com/pakrym/azure-sdk-for-net/blob/pakrym/update-generator-IOT/sdk/iot/Azure.Iot.Hub.Service/src/Generated/Models/ExportImportDevice.cs

Reply:
Correct.

Reply:
I see, what was the driving force behind this change? Should collections not be settable?

Reply:
The following guideline:

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections#collection-properties-and-return-values

> ❌ DO NOT provide settable collection properties.

> Users can replace the contents of the collection by clearing the collection first and then adding the new contents.

Reply:
This class can probably be internal. Will check and let you know. If it is internal, we would not need the extensions right?

Reply:
I'm not sure I understand why not. We would still need to create instances of this class and the code-gen for properties doesn't change depending on the class being internal or not.

Reply:
And I want to add here that this is still work in progress and we are yet to make some classes internal which we do not expose. Just curious, is there something you use to be able to detect these in the API surface?